packages/block-editor/src/components/grid/grid-visualizer.js

```diff
@@ -44,6 +45,28 @@ export function GridVisualizer( { clientId, contentRef, parentLayout } ) {
 	);
 }
 
+const checkIfCellOccupied = ( gridElement, column, row ) => {
+	const cell = Array.from( gridElement.children ).find( ( child ) => {
```

Comment:
Bit worried about performance here. We're iterating through the grid's children once per cell. So for a 16x16 grid with blocks in every cell that's `16*16*16*16` calls to `getComputedCSS` on every render.

An alternative approach might be to calculate `rects` similar to `use-grid-layout-sync.js:50` and then you can tell if a cell is occupied by looking for a rect where `rect.contains( row, column )`.

I might try that if you don't mind me pushing to this branch.

Reply:
Sure, go ahead!

Reply:
I think the bug you're seeing with appender still appearing when a paragraph is added must happening in this function, because the `gridElement` refreshes with the new paragraph and its correct col and row attributes, but `isCellOccupied` still returns false for that cell.

Reply:
How's this look? 9a626e3c6f

Not very scientific but it feels a bit snappier to me clicking around.

Reply:
Hmmm, I'm seeing `The 'useSelect' hook returns different values when called with the same state and parameters. This can lead to unnecessary rerenders. ` in the console when selecting the grid.

Reply:
But the issue with the appender still rendering is fixed! And the code looks much neater ☺️

Reply:
Oops. Fixed.